File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/api/callback/CodeRequestCallback.java
package cc.coopersoft.accesstoken.api.callback;

import java.util.Date;

public interface CodeRequestCallback {

  interface Result{
    long getExpiresIn();
    Date getCreated();
  }

  void onFailure(String phoneNumber, ErrorResult error);

  void onSuccess(String phoneNumber, Result result);
}

File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/api/callback/CodeRequestCallback.kt
package cc.coopersoft.accesstoken.api.callback

import java.util.Date

interface CodeRequestCallback {
    interface Result {
        val expiresIn: Long
        val created: Date?
    }

    fun onFailure(phoneNumber: String?, error: ErrorResult)
    fun onSuccess(phoneNumber: String?, result: Result)
}